YAML parser: Añadido soporte para expresiones booleanas

Kaian opened the pull request:
Este cambio, basado en la lógica de sustitucion actual de variables en los
ficheros de configuración de klear, permite incluir expresiones booleanas.

Las expresiones se deben encapsular entre los tags especiales $[ ] y se
evaluarán como una expresión en PHP, pudiendo devolver 'true' o 'false'.

Esta evaluación se hará **después** de la sustitución de variables, por lo que
la expressión puede contener variables sustituidas.

Esto puede ser especialmente práctico para evaluar blacklists, inclusiones de
botones o pantallas, etc en base a más de una variable, por ejemplo:

``` 
fields:
  blacklist:
    name: $[${auth.variable1} && ${auth.variable2}]
```

Aunque cualquier expressión que de devuelva un valor evaluable es válida

```
fields:
  screens:                                                              
     configurationEdit_screen: $[${auth.accessLevel} >=4]
```

Kaian committed:
YAML parser: Añadido soporte para expressiones booleanas

Este cambio, basado en la lógica de sustitucion actual de variables en los
ficheros de configuración de klear, permite incluir expresiones booleanas.

Las expresiones se deben encapsular entre los tags especiales $[ ] y se
evaluarán como una expresión en PHP, pudiendo devolver 'true' o 'false'.

Esta evaluación se hará **después** de la sustitución de variables, por lo que
la expressión puede contener variables sustituidas.

jabiinfante reviewed (changes requested):
Please, sanitize before eval.

Kaian commented:
Is there any other sanitizes in klear so I can copy the same behaviour?

PHP is not my jam

Thanks

jabiinfante commented:
Right here: https://github.com/irontec/klear/blob/chloe/models/Gettext.php

That method would only allow "ngettext","_" function calls.

Kaian committed:
YAML parser: Validate expression tokens before evaluation

Added a function that checks the tokens of the expression.

This function does not actually validate that the expression is correct, just
that the expression only contains allowed strings.

Kaian commented:
Heyo:

Basandome en la clase que me has pasado he creado una función que valida que los tokens de la expresión sean válidos. La función no comprueba que la expresión se correcta o léxicamente válida, tan solo que esta compuesta por strings validos.

He probado con varias expresiones del estilo `$[true and 1]`, `$[(1 >= 2) || (4 === true)]`, etc y parece que las acepta, mientras que rechaza cualquier otra que contenga palabras no permitidas en el listado de tokens.

Good shit el [token_get_all](http://php.net/manual/es/function.token-get-all.php)

Gracias!!
